go.mod

```diff
@@ -21,7 +21,7 @@ require (
 	github.com/go-telegram-bot-api/telegram-bot-api v4.6.4+incompatible
 	github.com/gregdel/pushover v1.3.0
 	github.com/hako/durafmt v0.0.0-20210608085754-5c1018a4e16b
-	github.com/hashicorp/nomad/api v0.0.0-20221121194306-f464aca721d2 // v1.4.3
+	github.com/hashicorp/nomad/api v0.0.0-20231212190655-0bc822db4037 // v1.4.3
```

Comment:
I think only this module needs to be updated right? If that's the case can you avoid updating unrelated modules? Thanks

Reply:
Also can we use a tag? Like https://github.com/hashicorp/nomad/releases/tag/v1.7.2?

Reply:
I think that doesn't work because the module is in a subdir of the repo. I get the following error:

```
$ go get -u github.com/hashicorp/nomad/api@v1.7.2
go: module github.com/hashicorp/nomad@v1.7.2 found, but does not contain package github.com/hashicorp/nomad/api
```

Is there some other way to annotate this?

As far as other modules being updated, `nomad/api` has dependencies that need to be updated as well, so it's more than just this one. One of those dependencies required bumping the version of another package to be compatible with an experimental interface change. I put this in the commit message to explain but didn't add it here after my rebase.

```
Update github.com/containers/image/v5

Makes compatible with changed API for slices.SortStableFunc.

https://github.com/containers/image/commit/336c7ff1189ba42d5c29a87dd5bed55e51608e9d
```

Reply:
> I think that doesn't work because the module is in a subdir of the repo. I get the following error:

Oh didn't catch it was a submodule. Then it should be this commit https://github.com/hashicorp/nomad/commit/64e3dca9274b493e38a49fda3a70fd31d0485b91 and add as comment in go.mod `v1.7.2` instead of `v1.4.3` so we are using a stable version.

> Update github.com/containers/image/v5

I don't see any dependency to `github.com/containers/image` or others you have updated here in https://github.com/hashicorp/nomad/blob/main/api/go.mod?

Reply:
Both depend on `/x/exp`. The newer version is required by nomad and includes a breaking API signature.

Reply:
Targeting the sha of v1.7.2 now.

Reply:
I checked on my side https://github.com/crazy-max/diun/pull/1056 and it doesn't require update of other modules. I guess you've done a `go get -u`?

Reply:
There are some other updates.

```
$ go get github.com/hashicorp/nomad/api@64e3dca
go: upgraded github.com/google/go-cmp v0.5.9 => v0.6.0
go: upgraded github.com/hashicorp/cronexpr v1.1.1 => v1.1.2
go: upgraded github.com/hashicorp/nomad/api v0.0.0-20221121194306-f464aca721d2 => v0.0.0-20231213195942-64e3dca9274b
go: upgraded golang.org/x/exp v0.0.0-20230522175609-2e198f4a06a1 => v0.0.0-20230728194245-b0cb94b80691
```

After that I vendored and tried to run tests and got the error below due to the updated `golang.org/x/exp`. So after that I had to 

```
$ docker buildx bake test
[+] Building 11.7s (16/17)
 => [internal] load build definition from Dockerfile                                                                                                         0.0s
 => => transferring dockerfile: 3.30kB                                                                                                                       0.0s
 => [internal] load .dockerignore                                                                                                                            0.0s
 => => transferring context: 160B                                                                                                                            0.0s
 => resolve image config for docker.io/docker/dockerfile:1                                                                                                   0.3s
 => CACHED docker-image://docker.io/docker/dockerfile:1@sha256:ac85f380a63b13dfcefa89046420e1781752bab202122f8f50032edf31be0021                              0.0s
 => [internal] load metadata for docker.io/library/golang:1.21-alpine                                                                                        0.3s
 => [internal] load metadata for docker.io/tonistiigi/xx:1.2.1                                                                                               0.2s
 => [xx 1/1] FROM docker.io/tonistiigi/xx:1.2.1@sha256:8879a398dedf0aadaacfbd332b29ff2f84bc39ae6d4e9c0a1109db27ac5ba012                                      0.0s
 => [internal] load build context                                                                                                                            0.2s
 => => transferring context: 9.88MB                                                                                                                          0.2s
 => [base 1/4] FROM docker.io/library/golang:1.21-alpine@sha256:feceecc0e1d73d085040a8844de11a2858ba4a0c58c16a672f1736daecc2a4ff                             0.0s
 => CACHED [base 2/4] COPY --from=xx / /                                                                                                                     0.0s
 => CACHED [base 3/4] RUN apk add --no-cache file git                                                                                                        0.0s
 => CACHED [base 4/4] WORKDIR /src                                                                                                                           0.0s
 => [vendored 1/2] COPY go.mod go.sum ./                                                                                                                     0.0s
 => [vendored 2/2] RUN --mount=type=cache,target=/go/pkg/mod   go mod download                                                                               0.2s
 => [test 1/2] RUN apk add --no-cache gcc linux-headers musl-dev                                                                                             3.3s
 => ERROR [test 2/2] RUN --mount=type=bind,target=.     --mount=type=cache,target=/go/pkg/mod     --mount=type=cache,target=/root/.cache/go-build <<EOT (se  7.2s
------
 > [test 2/2] RUN --mount=type=bind,target=.     --mount=type=cache,target=/go/pkg/mod     --mount=type=cache,target=/root/.cache/go-build <<EOT (set -ex...):
#0 0.361 + go test -parallel 4 -v '-coverprofile=/tmp/coverage.txt' '-covermode=atomic' -race ./...
#0 0.801 # github.com/containers/image/v5/internal/manifest
#0 0.801 /go/pkg/mod/github.com/containers/image/v5@v5.26.1/internal/manifest/oci_index.go:154:42: type func(a v1.Descriptor, b v1.Descriptor) bool of func(a, b imgspecv1.Descriptor) bool {…} does not match inferred type func(a v1.Descriptor, b v1.Descriptor) int for func(a E, b E) int
#0 5.728 ?   	github.com/crazy-max/diun/v4/internal/grpc/logger	[no test files]
#0 5.728 ?   	github.com/crazy-max/diun/v4/internal/logging	[no test files]
#0 5.729 FAIL	github.com/crazy-max/diun/v4/internal/config [build failed]
#0 5.729 FAIL	github.com/crazy-max/diun/v4/internal/provider [build failed]
#0 5.729 ?   	github.com/crazy-max/diun/v4/pb	[no test files]
#0 5.729 ?   	github.com/crazy-max/diun/v4/pkg/docker	[no test files]
#0 5.729 FAIL	github.com/crazy-max/diun/v4/internal/provider/file [build failed]
#0 5.729 FAIL	github.com/crazy-max/diun/v4/internal/provider/nomad [build failed]
#0 5.991 ?   	github.com/crazy-max/diun/v4/pkg/k8s	[no test files]
#0 5.991 ?   	github.com/crazy-max/diun/v4/pkg/utl	[no test files]
#0 7.016 === RUN   TestNew
#0 7.016 --- PASS: TestNew (0.00s)
#0 7.016 === RUN   TestLoadFile
#0 7.016 === RUN   TestLoadFile/Failed_on_non-existing_file
#0 7.016 cannot read Dockerfile : open : no such file or directory
#0 7.016 === RUN   TestLoadFile/Fail_on_empty_file
#0 7.016 cannot parse Dockerfile ./fixtures/empty.Dockerfile: file with no instructions
#0 7.016 === RUN   TestLoadFile/Fail_on_wrong_file_format
#0 7.016 cannot parse stages for Dockerfile ./fixtures/invalid.Dockerfile: dockerfile parse error on line 1: unknown instruction: invalid
#0 7.016 === RUN   TestLoadFile/Valid
#0 7.016 --- PASS: TestLoadFile (0.00s)
#0 7.016     --- PASS: TestLoadFile/Failed_on_non-existing_file (0.00s)
#0 7.016     --- PASS: TestLoadFile/Fail_on_empty_file (0.00s)
#0 7.016     --- PASS: TestLoadFile/Fail_on_wrong_file_format (0.00s)
#0 7.016     --- PASS: TestLoadFile/Valid (0.00s)
#0 7.016 === RUN   TestFromImages
#0 7.016 --- PASS: TestFromImages (0.00s)
#0 7.016 PASS
#0 7.016 coverage: 84.3% of statements
#0 7.016 ok  	github.com/crazy-max/diun/v4/pkg/dockerfile	1.023s	coverage: 84.3% of statements
#0 7.016 FAIL	github.com/crazy-max/diun/v4/pkg/registry [build failed]
#0 7.016 FAIL
------
Dockerfile:34
--------------------
  33 |     RUN apk add --no-cache gcc linux-headers musl-dev
  34 | >>> RUN --mount=type=bind,target=. \
  35 | >>>     --mount=type=cache,target=/go/pkg/mod \
  36 | >>>     --mount=type=cache,target=/root/.cache/go-build <<EOT
  37 | >>>   set -ex
  38 | >>>   go test -parallel 4 -v -coverprofile=/tmp/coverage.txt -covermode=atomic -race ./...
  39 | >>>   go tool cover -func=/tmp/coverage.txt
  40 | >>> EOT
  41 |
--------------------
ERROR: failed to solve: process "/bin/sh -c   set -ex\n  go test -parallel 4 -v -coverprofile=/tmp/coverage.txt -covermode=atomic -race ./...\n  go tool cover -func=/tmp/coverage.txt\n" did not complete successfully: exit code: 1
```

So that's why I need to update containers/images.

This time I did it without `-u` and it updated a bit less, but still more than the one dependency.